variables.tf

```diff
@@ -217,6 +217,18 @@ variable "kubelet_additional_options" {
   }
 }
 
+variable "ami_name_regex" {
+  type        = string
+  description = "Override the name regex generated by ami_type"
+  default     = []
```

Comment:
I believe it's a list in order to avoid for_each/count issues. All of the strings in this module are exposed as single entry lists.

cc: @Nuru might have more context for that

Reply:
@osterman see https://github.com/cloudposse/terraform-aws-eks-node-group/blob/780163dacd9c892b64b988077a994f6675d8f56d/MIGRATION.md

> The following optional values used to be string type and are now list(string) type. An empty list is allowed. If the list has a value in it, that value will be used, even if empty, which may not be allowed by Terraform. The list may not have more than one value.

Reply:
also see https://github.com/cloudposse/terraform-aws-eks-node-group/blob/780163dacd9c892b64b988077a994f6675d8f56d/MIGRATION.md#convert-optional-variables-to-lists

> The biggest number of changes is in the optional variables. We used to determine whether or note a variable was set by looking at its value: null or the empty string was "not set" and any other value was "set". Unfortunately, Terraform does not work that way. So we now take optional variables as a list with zero or one item. If the list is empty (zero items), the variable is not set; if the list has an item, the variable is set.

Reply:
Oh woops, I forgot to change it to `type = list(string)`. Corrected this.